core/src/test/scala/mill/main/ParseArgsTest.scala

```diff
@@ -0,0 +1,290 @@
+package mill.main
+
+import mill.discover.Mirror
+import utest._
+
+object ParseArgsTest extends TestSuite {
+
+  val emptyArgs = Seq.empty[String]
+
+  val singleSelector = Seq("core.compile")
+
+  val singleSelectorWithCross = Seq("bridges[2.12,jvm].compile")
+
+  val singleSelectorWithArgs = Seq("application.run", "hello", "world")
+
+  val multiSelectorsAll =
+    Seq("--all", "core.jar", "core.docsJar", "core.sourcesJar")
+
+  val multiSelectorsSeq =
+    Seq("--seq", "core.jar", "core.docsJar", "core.sourcesJar")
+
+  val multiSelectorsWithArgsAll = Seq("--all",
+                                      "core.compile",
+                                      "application.runMain",
+                                      "--",
+                                      "Main",
+                                      "hello",
+                                      "world")
+
+  val multiSelectorsBraceExpansion = Seq("--all", "{core,application}.compile")
+
+  val multiSelectorsBraceExpansionWithCross =
+    Seq("--all", "bridges[2.12,jvm].{test,jar}")
+
+  val multiSelectorsBraceExpansionWithArgs =
+    Seq("--all", "{core,application}.run", "--", "hello", "world")
+
+  val tests = Tests {
+    'extractSelsAndArgs - {
```

Comment:
All these tests can be DRYed up with a helper method 
```scala
'empty - check(emptyArgs, Nil, Nil)
'singleSelector - check(singleSelector, singleSelector, empty)
'singleSelectorWithArgs - check(singleSelectorWithArgs, Seq("application.run"), Seq("hello", "world"))
```

With an appropriate definition for `check`. The tests under `'expandBraces` and `'apply` seem like they can also be DRYed up with the appropriate helper method